datafusion/physical-expr/src/aggregate/count_distinct/mod.rs

```diff
@@ -268,8 +268,11 @@ impl Accumulator for DistinctCountAccumulator {
         let array = &states[0];
         let list_array = array.as_list::<i32>();
         for inner_array in list_array.iter() {
```

Comment:
should we just filter out Nones? just thinking

Reply:
I think filtering out nulls would potentially silently ignore errors

I think the rationale goes like
1. This code is used to combine multiple partial results (the results of calling `Self::state()`)
2. Since `Self::state` never returns `None` -- it always returns `Ok(vec![ScalarValue::List(arr)])` then this merge code should never receive a `None`